gym/envs/classic_control/cartpole.py

```diff
@@ -155,12 +155,14 @@ def step(self, action):
 
         return np.array(self.state, dtype=np.float32), reward, done, {}
 
-    def reset(self, seed: Optional[int] = None):
+    def reset(self, seed: Optional[int] = None, return_info: Optional[bool] = False):
         super().reset(seed=seed)
         self.state = self.np_random.uniform(low=-0.05, high=0.05, size=(4,))
         self.steps_beyond_done = None
-        return np.array(self.state, dtype=np.float32)
-
+        if not return_info:
```

Comment:
So this will be the problematic part, since we'll need to patch this in all environments (not a big deal), but also make sure that third-party environments won't have problems with it (and I don't think there's anything we can do to automatically make them obey the new contract of `reset`)

Reply:
I've started work on extending this patch to all first-party environments, so far, I haven't found any which return non-empty metadata, which is interesting.